packages/main/src/Table.ts

```diff
@@ -464,11 +465,19 @@ class Table extends UI5Element {
 	}
 
 	get styles() {
-		return {
+		const headerStyleMap = this.headerRow?.[0]?.cells?.reduce((headerStyles, headerCell) => {
+			if (headerCell.horizontalAlign !== undefined && Object.values(TableCellHorizontalAlign).includes(headerCell.horizontalAlign as TableCellHorizontalAlign)) {
```

Comment:
horizontalAlign should be compared with default value (Start)
why do we need the 2nd part of the check?